baseplate/file_watcher.py

```diff
@@ -63,15 +64,59 @@ class FileWatcher(object):
     :param callable parser: A callable that takes an open file object, parses
         or otherwise interprets the file, and returns whatever data is
         meaningful.
-    :param float timeout: How long, in seconds, to block instantiation waiting
-        for the watched file to become available (defaults to not blocking).
+    :param float timeout: (Optional) How long, in seconds, to block
+        instantiation waiting for the watched file to become available
+        (defaults to not blocking).
+    :param bool binary: (Optionaly) Should the file be opened in binary mode. If
+        `True` the file will be opened with the mode "r+b", otherwise it will be
+        opened with the mode "r". (defaults to "r")
+    :param str encoding: (Optional) The name of the encoding used to decode
+        the file. The default encoding is platform dependent (whatever
+        locale.getpreferredencoding() returns), but any text encoding supported
+        by Python can be used.  This is not supported in Python 2 or if `binary`
+        is set to `True`.
+    :param str newline: (Optional) Controls how universal newlines mode works
+        (it only applies to text mode). It can be `None`, `''`, `'\n'`, `'\r'`,
+        and `'\r\n'`.  This is not supported in Python 2 or if `binary` is set
+        to `True`.
 
     """
-    def __init__(self, path, parser, timeout=None):
+    def __init__(self, path, parser, timeout=None, binary=False, encoding=None,
+                 newline=None):
+        if sys.version_info.major < 3 and encoding is not None:
+            raise TypeError(
+                "'encoding' keyword argument for FileWatcher() is not "
+                "supported in Python 2"
+            )
+
+        if sys.version_info.major < 3 and newline is not None:
+            raise TypeError(
+                "'newline' keyword argument for FileWatcher() is not "
+                "supported in Python 2"
+            )
+
+        if binary and encoding is not None:
+            raise TypeError("'encoding' is not supported in binary mode.")
+
+        if binary and newline is not None:
+            raise TypeError("'newline' is not supported in binary mode.")
+
         self._path = path
         self._parser = parser
         self._mtime = 0
         self._data = _NOT_LOADED
+        self._mode = "r+b" if binary else "r"
```

Comment:
`r+b` is opening the file for append-writes and reads in binary mode rather than just `rb` which is "read only binary mode"